jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ServletContextListenerTest.java

```diff
@@ -0,0 +1,163 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.servlet;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.handler.HotSwapHandler;
+import org.eclipse.jetty.toolchain.test.FS;
+import org.eclipse.jetty.toolchain.test.MavenTestingUtils;
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.eclipse.jetty.util.resource.PathResource;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+
+public class ServletContextListenerTest
+{
+    private Server server;
+
+    @AfterEach
+    public void tearDown()
+    {
+        LifeCycle.stop(server);
+    }
+
+    private Server newServer()
+    {
+        server = new Server();
+        ServerConnector connector = new ServerConnector(server);
+        connector.setPort(0);
+        server.addConnector(connector);
+        return server;
+    }
+
+    @Test
+    public void testServletContextListener() throws Exception
+    {
+        Server server = newServer();
+        HotSwapHandler swap = new HotSwapHandler();
+        server.setHandler(swap);
+        server.start();
+
+        Path tempDir = MavenTestingUtils.getTargetTestingPath("testServletContextListener");
+        FS.ensureEmpty(tempDir);
+
+        ServletContextHandler context = new ServletContextHandler(
+            ServletContextHandler.SESSIONS);
+        context.setContextPath("/");
+        context.setBaseResource(new PathResource(tempDir));
+
+        final List<String> history = new ArrayList<>();
+
+        context.addEventListener(new ServletContextListener()
+        {
+            @Override
+            public void contextInitialized(ServletContextEvent servletContextEvent)
+            {
+                history.add("I0");
```

Comment:
Theses tests came from WebAppContextTest.

https://github.com/eclipse/jetty.project/blob/jetty-9.4.31.v20200723/jetty-webapp/src/test/java/org/eclipse/jetty/webapp/WebAppContextTest.java#L314-L415

Since they didn't test WebAppContext in any fashion (in fact it only uses ServletContextHandler), I moved them out to a new testcase.

Reply:
Sure.... but we still need to test adding a `ServletContextListener` from another `ServletContextListener`.   It seams this test it the correct place to test that.... even if we didn't test it before.

Reply:
Have a look at the https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ServletContextHandlerTest.java.  It does a lot of tests of adding listeners from listeners, servlets from listeners etc etc, maybe this should be in there, or is already covered in there.

Reply:
This test has essentially duplicated tests from what exists in ServletContextHandlerTest.
This class has been removed in this PR.